### ISSUE TITLE
Tensorflow prediction keeps returning a sequence of the same numbers 

### ISSUE THREAD

**Issue by AaronWard** [SurveNet/model-training-tensorflow#1]
I feed in a random sample of 10 test data images and most of the time (probably 9 times out of 10) i returns the number sequence of the same number, like [ 2 2 2 2 2 2 2 2 2 2] or [1 1 1 1 1 1 1 1 1 1 1 ]

```
def test_network(x, images, labels):
    sample_images, sample_labels = random_sample(images, labels)
    sample_images = np.array(sample_images).astype(float)

    pred = convolutional_network(x)
    correct_pred = tf.argmax(pred, 1)

    with tf.Session() as sess: 
        sess.run(tf.global_variables_initializer())
        saver = tf.train.import_meta_graph(TRAINED_MODEL + 'trained_model.ckpt.meta')
        saver.restore(sess, TRAINED_MODEL + 'trained_model.ckpt' )
        print('session restored...')
        
   #Here is where i feed the 10 random images
        predicted = sess.run([correct_pred], feed_dict={x: sample_images})[0]


        print('Correct Labels: ', sample_labels)
        print('Predicted Labels: ', predicted)

        
        #Plot the images samples in a grid with the predicted
        #outputs and the true outputs
        fig = plt.figure(figsize=(10, 10))
        for i in range(len(sample_images)):
            truth = sample_labels[i]
            prediction = predicted[i]
            plt.subplot(5, 2,1+i)
            plt.axis('off')
            color='green' if truth == prediction else 'red'
            plt.text(60, 10, "Correct:        {0}\nPrediction: {1}".format(truth, prediction), 
                     fontsize=12, color=color)
            plt.imshow(sample_images[i], cmap='gray')
        plt.show() 
     
test_network(x, images, labels)
```

![22222222](https://user-images.githubusercontent.com/22074246/38895996-706239ac-4289-11e8-8514-1aaae7c8a925.JPG)

## SOLUTION
```
it's happening because of correct_pred = tf.argmax(pred, 1), it's giving you 
class with the highest probability after the softmax. You can use predicted
 = sess.run(pred, feed_dict={x: test_images[0:10]}). Now, you will get 
probability of each class for given images For example you might 
get [.1, .05, .05 , .6, .1, .1] for your 6 classes. You won't get [0,0,0,1,0,0].
Now, argmax will give you index corresponding to .6.

Also, add :
pred_ = tf.nn.softmax(pred)
predicted = sess.run(pred_, feed_dict={x: test_images[0:10]})
```

**Comment by pshah123** [comment]
Images don't look normalized--at the least I recommend dividing by 255. Also, what does your graph look like? If you have ReLU switch to Leaky ReLU, make sure your data is between the input range for the function also.

**Comment by AaronWard** [comment]
@pshah123 why should i divide by 255 may i ask? 

Would you be able to have a quick look at my scripts and see if you can find anything our of place? 
I have trained the model on Floydhub, and now i am testing locally by loading the trained model into the evaluation script. 

#Training script:
https://github.com/SurveNet/model-training-tensorflow/blob/master/notebooks/Floydhub%20Notebooks/train.ipynb

# Evaluation Script (the one i'm having trouble with): 
https://github.com/SurveNet/model-training-tensorflow/blob/master/notebooks/single_prediction.ipynb
